mods/page_save.js

```diff
@@ -124,16 +165,23 @@ PageSave.prototype.saveHtml = function(restbase, req) {
     var self = this;
     var rp = req.params;
     var title = rbUtil.normalizeTitle(rp.title);
-    var promise = P.resolve({});
     this._checkParams(req.body);
     // First transform the HTML to wikitext via the parsoid module
+    var path = [rp.domain, 'sys', 'parsoid', 'transform', 'html', 'to', 'wikitext', title];
+    var baseRevision = this._getBaseRevision(req);
+    if (baseRevision) {
+        path.push(baseRevision);
+    }
     return restbase.post({
-        uri: new URI([rp.domain, 'sys', 'parsoid', 'transform', 'html', 'to', 'wikitext', title]),
+        uri: new URI(path),
         body: {
-            revision: req.body.revision,
             html: req.body.html
+        },
+        headers: {
+            'if-match': req.body.base_etag || req.headers['if-match']
```

Comment:
Ah, no, it should be ok, since now the transform endpoints require a tid to be supplied. Disregard.